runtime/util/hshelp.c

```diff
@@ -2701,59 +2701,62 @@ addClassesRequiringNewITables(J9JavaVM *vm, J9HashTable *classHashTable, UDATA *
 				result = hashTableFind(classHashTable, &exemplar);
 
 				if (NULL != result) {
-					BOOLEAN extensionsUsed = (0 != (result->flags & J9JVMTI_CLASS_PAIR_FLAG_USES_EXTENSIONS));
-
-					/* If methods were added or removed (extensionsUsed) or re-ordered in the interface, the class should be redefined. */
-					if (extensionsUsed || (NULL != result->methodRemap)) {
-						U_32 nodeCount = hashTableGetCount(classHashTable);
-
-						memset(&exemplar, 0, sizeof(J9JVMTIClassPair));
-						exemplar.originalRAMClass = clazz;
-						exemplar.replacementClass.romClass = clazz->romClass;
-						exemplar.flags = J9JVMTI_CLASS_PAIR_FLAG_REDEFINED;
-
-						/* An interface of this class is being updated with extensions - this class will require new iTables. */
-						if (NULL == hashTableAdd(classHashTable, &exemplar)) {
-							vmFuncs->allClassesEndDo(&classWalkState);
-							return JVMTI_ERROR_OUT_OF_MEMORY;
+					UDATA flags = 0;
+					U_32 nodeCount = hashTableGetCount(classHashTable);
+
+					if (!fastHCR) {
+						BOOLEAN extensionsUsed = (0 != (result->flags & J9JVMTI_CLASS_PAIR_FLAG_USES_EXTENSIONS));
+						/* If methods were added or removed (extensionsUsed) or re-ordered in the interface, the class should be redefined. */
+						if (extensionsUsed || (NULL != result->methodRemap)) {
+							flags = J9JVMTI_CLASS_PAIR_FLAG_REDEFINED;
 						}
+					}
 
-						/* Increment counts if the class was really added (i.e. did not exist before). */
-						if (hashTableGetCount(classHashTable) == (nodeCount + 1)) {
-							J9SubclassWalkState subclassState;
-							J9Class * subclass;
+					memset(&exemplar, 0, sizeof(J9JVMTIClassPair));
+					exemplar.originalRAMClass = clazz;
+					exemplar.replacementClass.romClass = clazz->romClass;
+					exemplar.flags = flags;
 
-							addedMethodCount += clazz->romClass->romMethodCount;
-							addedClassCount += 1;
-	
-							/* Now add all of the subclasses of the affected class to the table */
-							subclass = allSubclassesStartDo(clazz, &subclassState, TRUE);
-							while (subclass != NULL) {
-								U_32 nodeCount = hashTableGetCount(classHashTable);
-
-								memset(&exemplar, 0x00, sizeof(J9JVMTIClassPair));
-								exemplar.originalRAMClass = subclass;
-								exemplar.replacementClass.romClass = subclass->romClass;
-								exemplar.flags = J9JVMTI_CLASS_PAIR_FLAG_REDEFINED;
-
-								/* If this class is already in the table this add will have no effect */
-								result = hashTableAdd(classHashTable, &exemplar);
-								if (NULL == result) {
-									vmFuncs->allClassesEndDo(&classWalkState);
-									return JVMTI_ERROR_OUT_OF_MEMORY;
-								}
-								/* Ensure that classes already in the table get redefined */
-								result->flags |= J9JVMTI_CLASS_PAIR_FLAG_REDEFINED;
+					/* An interface of this class is being updated with extensions - this class will require new iTables. */
```

Comment:
Is this comment still accurate?  Maybe it needs to move up and into the `if (!fastHCR) {` block?

Reply:
Moved up.